components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/util/EventPayloadCryptographyUtils.java

```diff
@@ -0,0 +1,195 @@
+                 IllegalBlockSizeException | BadPaddingException | ParseException e) {
+            throw new IdentityEventException("Error while encrypting event payload.", e);
+        }
+    }
+
+    private static PublicKey getPublicKey(String tenantDomain) throws ParseException, InvalidKeyException,
+            IdentityEventException, JOSEException {
+
+        try {
+            DefaultJWKSetCache orgJWKCache = getJWKCache(tenantDomain);
+            if (orgJWKCache.get() != null && !orgJWKCache.isExpired()) {
+                return convertJWKToPublicKey(orgJWKCache.get());
+            }
+            synchronized (orgJWKCache) {
+                // Recheck cache for a valid key once the lock is obtained as another thread could have updated the
+                // cache while this was waiting.
+                if (orgJWKCache.get() != null && !orgJWKCache.isExpired()) {
+                    return convertJWKToPublicKey(orgJWKCache.get());
+                }
+
+                Resource keyResource = retrieveKeyFromAPI(tenantDomain);
+                JSONParser jsonParser = new JSONParser();
+                JSONObject responseJSON = (JSONObject) jsonParser.parse(keyResource.getContent());
+                if (responseJSON.get(CRYPTO_KEY_RESPONSE_JSON_KEY) == null) {
+                    throw new IdentityEventException("Event encryption public key endpoint has returned an " +
+                            "invalid response.");
+                }
+                byte[] publicKeyBytes =
+                        Base64.getDecoder().decode(responseJSON.get(CRYPTO_KEY_RESPONSE_JSON_KEY).toString());
+                RSAPublicKey rsaPublicKey = RSAPublicKeyImpl.newKey(publicKeyBytes);
+                RSAKey compositePublicKeyJWK = new RSAKey(Base64URL.encode(rsaPublicKey.getModulus()),
+                        Base64URL.encode(rsaPublicKey.getPublicExponent()), null, null, null, null, null,
+                        null, null, null, KeyUse.ENCRYPTION, null, JWEAlgorithm.RSA_OAEP_256,
+                        null, null, null, null, null, null);
+                JWKSet jwkSet = new JWKSet(Collections.singletonList(compositePublicKeyJWK));
+                orgJWKCache.put(jwkSet);
+                return convertJWKToPublicKey(jwkSet);
+            }
+        } catch (IOException e) {
+            throw new IdentityEventException("Unable to fetch event encryption public key for tenant " +
+                    tenantDomain, e);
+        }
+    }
+
+    private static DefaultJWKSetCache getJWKCache(String tenantDomain) {
+
+        if (cacheMap.get(tenantDomain) == null) {
+            int cacheLifeSpan = WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                    .getEncryptionKeyCacheLifespan();
+            cacheMap.putIfAbsent(tenantDomain, new DefaultJWKSetCache(cacheLifeSpan, TimeUnit.MINUTES));
+        }
+        return cacheMap.get(tenantDomain);
```

Comment:
FYI: you can use `map.computeIfAbsent()` in cases like this, if found return existing or return the new value added.
```suggestion
        return cacheMap.computeIfAbsent(tenantDomain, new DefaultJWKSetCache(WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
                    .getEncryptionKeyCacheLifespan(), TimeUnit.MINUTES));
```

Reply:
Thank you for the suggestion, Seem we can use putIfAbsent (Instead of computeIfAbsent) as the second argument we are passing here is not a function.  
+1 to proceed? @wenodpathirana